httpclient5/src/main/java/org/apache/hc/client5/http/SystemDefaultDnsResolver.java

```diff
@@ -44,8 +45,9 @@ public class SystemDefaultDnsResolver implements DnsResolver {
     @Override
     public InetAddress[] resolve(final String host) throws UnknownHostException {
         try {
+            final String normalizedHost = IDN.toASCII(host);
```

Comment:
@arturobernalg I would do  `IDN.toASCII` only if the hostname fails `TextUtils#isAllASCII` for the sake of optimization.

Reply:
@arturobernalg Catching `IllegalArgumentException` and recovering from it (for instance, by keeping the original hostmame) would be nice here

Reply:
@ok2c  Please do another pass

Reply:
@arturobernalg Do you have any objections to doing the test `TextUtils#isAllASCII` prior to doing `IDN.toASCII`?

Reply:
@ok2c done